src/Jenssegers/Mongodb/Relations/BelongsToMany.php

```diff
@@ -171,7 +185,30 @@ public function sync($ids, $detaching = true)
      */
     public function updateExistingPivot($id, array $attributes, $touch = true)
     {
-        // Do nothing, we have no pivot table.
+        if ($id instanceof Model) {
+            $model = $id;
+            $id = $model->getKey();
+        } else {
+            if ($id instanceof Collection) {
+                $id = $id->modelKeys();
```

Comment:
Maybe better to rename `$id` to `$idModelKeys` and after line to cast into array

Reply:
Well, there is an `parseIds` function which is doing that, not sure why we're doing double work.

https://github.com/laravel/framework/blob/dc2b6cca647044a7041616ea3e45e27b9123d0d8/src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php#L571

Reply:
I think that author of parent PR don't see this function

Reply:
Maybe use this function and remove this code?

Reply:
Well, I don't like this implementation at all, it has a tests but implementation isn't following framework conventions, I think it needs complete rewrite from scratch, maybe close this at all?